avaframe/com4FlowPy/flowCore.py

```diff
@@ -361,6 +364,7 @@ def calculation(args):
     startcell_idx = 0
     while startcell_idx < len(row_list):
 
+        processedCells = {}  # dictionary of cells that have been processed already
```

Comment:
If we used a list here, I think we could change the part where we search, if the cell already exists (line 405), that is now quite inefficient. I've tried that in the [PS_flowpy_fluxSum](https://github.com/avaframe/AvaFrame/tree/PS_flowpy_fluxSum) branch.
Just an idea for the next time :)

Reply:
@PaulaSp3 if you have an idea like this, feel free to start a new issue from here. I added a new label called flowPyDev, just tag the issue with that...

Reply:
I just use this dictionary as a lookup table for now, i.e. every time a cell is added to the cell_list, where the actual processing is taking place, I look if this cell (col, row) is already in the dict - if not, then the cell is added with a counter of 1, if the cell has been 'visited' already this counter is incremented.

- The lookup if an element is in a list is O(n)
- The lookup if an element is in a dictionary is O(1)

so using a list for this specific purpose would be slower imo

Reply:
I also tried using a dictionary for ```cell_list```, but did not manage to produce the same results as the list implementation $\rightarrow$ But we can discuss this :)